tower_cli/cli/transfer/send.py

```diff
@@ -1235,7 +1235,7 @@ def process_group_comparison(self, existing_group, new_group, existing_group_ids
             for host_name in (set(existing_hosts) | set(new_hosts)):
                 # Get the host object to get the ID
                 try:
-                    host = tower_cli.get_resource('host').get(**{'name': host_name})
+                    host = tower_cli.get_resource('host').get(**{'name': host_name, 'inventory': inventory['id']})
```

Comment:
this would be the "new" inventory, I take it

Reply:
Correct, so when we try to associate to disassociate a host we are checking for hosts in the inventory we are working on, not just a host in general. This is what was causing it to pickup hosts with the same name from other inventories.